### DISCUSSION TITLE
Getting this error but I am sure it is uuid.  operator does not exist: uuid = text

### DISCUSSION THREAD

**Discussion by tarkanlar** [supabase/supabase#11950] [Questions] [answered]
Hi everyone, 

isUserUUID always returns true and in profiles table id is uuid but I get this error.

` {"code": "42883", "details": null, "hint": "No operator matches the given name and argument types. You might need to add explicit type casts.", "message": "operator does not exist: uuid = text"}`

and this is my code:

```
if (session !== null && session.user.id !== null) {

const regexExp = /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/gi;

const isUserIdUUID = regexExp.test(session.user.id); 

console.log("isUserIdUUID", isUserIdUUID)

const profile = await supabase.from("profiles").select("*").eq("id", session.user.id).single()

 if (profile.data !== null) 
setProfile(profile.data)
else if (profile.error !== null)
console.log('this--------', profile.error)
                    }

```

**Comment by GaryAustin1** [comment, marked as answer]
Try and restart your database if you ever changed that column from uuid/text or opposite.   There is a bug with caching.
https://github.com/supabase/supabase/issues/11714

**Comment by tarkanlar** [comment]
Thank you so much I spent half day while trying to figure out
